MozillaFoundation/foundation.mozilla.org
# Remove unwanted gaps between hero and footer when using dark sections

# Description

The problem is that Issue #11756 introduced unwanted white gaps in between the dark hero and an immediately following dark section as seen by screenshots in Issue #11795 . Additionally, there is an unwanted white gap between the body section and the footer when the page ends with a dark section.

The solution is to conditionally add spacing above a paragraph block when it immediately follows a dark hero instead of after the dark hero unconditionally. The solution for the second unwanted white gap is to simply remove the margin class from the mozfest-content wrapper.

The implementation introduces a streamfield custom block which will be overridden to include a new streamfield_spacer.html template on mozfest_homepage.html and mozfest_landing_page.html templates. This new template extends the existing streamfield.html template to include a new body_spacer custom block containing a spacer div. The body_spacer block will then be rendered only if a paragraph is the first block in the body section. The body_spacer block is empty by default, meaning the spacer div is only rendered when the two following conditions are met:

1. Mozfest Homepage or Mozfest Landing Page
2. First block in body is a paragraph

There are further considerations. We could reduce coupling by adding margins to the paragraph block unconditionally to have adequate spacing in all cases regardless of placement. But the paragraph block is used generically and currently assumes spacing from blocks above it (see for example the various uses on just this one page: https://www.mozillafestival.org/en/highlights/). Modifying each dependency in order to introduce a generic paragraph margin seems out of scope for this issue, but perhaps could be a larger follow-up issue.

Link to sample test page: https://mozfest-foundation-s-issue-1179-lpznkg.mofostaging.net/en/test-landing/
Related PRs/issues: #11756 and #11795

# Steps to test

1. Review the dark section example page to confirm the white spaces are removed: https://mozfest-foundation-s-issue-1179-lpznkg.mofostaging.net/en/test-landing/
2. Review the landing page example page to confirm the margin / spacer div is present above the first paragraph block and below the hero: https://mozfest-foundation-s-issue-1179-lpznkg.mofostaging.net/en/landing-page-1/
3. Review another page to confirm the spacing is still accurate / untouched: https://mozfest-foundation-s-issue-1179-lpznkg.mofostaging.net/en/tickets/
